File: 240806/최고의 13위치/best-place-of-13.py
N = int(input())

arr = [list(map(int, input().split())) for _ in range(N)]

max_count = 0

for i in range(N):
    for j in range(N-2):
        count = 0
        for k in range(3):
            if arr[i][j+k] == 1:
                count += 1
        max_count = max(max_count, count)

print(max_count)

File: 240815/최고의 13위치/best-place-of-13.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();

        int arr[][] = new int[N][N];

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                arr[i][j] = sc.nextInt();
            }
        }

        int max_count = 0;
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N-2; j++) {
                int count = 0;
                for (int k = 0; k < 3; k++) {
                    if (arr[i][j+k] == 1) {
                        count++;
                    }
                }
                max_count = Math.max(max_count, count);
            }
        }

        System.out.print(max_count);

    }
}